scio-parquet/src/main/java/com/spotify/scio/parquet/avro/ParquetAvroFileBasedSink.java

```diff
@@ -107,9 +111,21 @@ public ParquetAvroWriter(
     @Override
     protected void prepareWrite(WritableByteChannel channel) throws Exception {
       BeamOutputFile outputFile = BeamOutputFile.of(channel);
+      Configuration configuration = conf.get();
+
+      // Workaround for PARQUET-2265
+      Class<? extends AvroDataSupplier> dataModelSupplier =
+          configuration.getClass(
+              AvroWriteSupport.AVRO_DATA_SUPPLIER,
+              SpecificDataSupplier.class,
+              AvroDataSupplier.class);
+
       AvroParquetWriter.Builder<T> builder =
-          AvroParquetWriter.<T>builder(outputFile).withSchema(schema);
-      writer = WriterUtils.build(builder, conf.get(), compression);
+          AvroParquetWriter.<T>builder(outputFile)
+              .withSchema(schema)
+              .withDataModel(ReflectionUtils.newInstance(dataModelSupplier, configuration).get());
```

Comment:
This is directly copied from parquet's [AvroWriteSupplier](https://github.com/apache/parquet-mr/blob/59e9f78b8b3a30073db202eb6432071ff71df0ec/parquet-avro/src/main/java/org/apache/parquet/avro/AvroWriteSupport.java#L403-L407); unfortunately the method is private so I can't just invoke it.

Reply:
anyway we can add test coverage for this?